codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java

```diff
@@ -1550,6 +1556,27 @@ public HttpHeaders set(HttpHeaders headers) {
     }
 
     /**
+     * Retains all current headers but calls {@link #set(String, Object)} for each entry in {@code headers}
+     *
+     * @param headers The headers used to {@link #set(String, Object)} values in this instance
+     * @return {@code this}
+     */
+    public HttpHeaders setAll(HttpHeaders headers) {
+        if (headers == null) {
+            throw new NullPointerException("headers");
+        }
+
+        if (headers.isEmpty()) {
```

Comment:
Fixed. I have also updated `HttpHeaders.set()` to use `forEachEntry()` as is done by `DefaultHeaders`

I also added a set of set/addShort() methods matching what we have in `DefaultHeaders`. `setShort()` was required by `InboundHttp2ToHttpPriorityAdapter`